core/state/statedb_evm.go

```diff
@@ -95,7 +94,7 @@ func (s *Statedb) Suicide(address common.Address) bool {
 	}
 
 	stateObj.SetAmount(new(big.Int))
-	// @todo mark the state object as suicided
+	stateObj.suicided = true
```

Comment:
do we need to add check to verify whether this account already suicided?

Reply:
just mark a state object as suicided, and physically delete from trie/DB when commit the statedb DB.